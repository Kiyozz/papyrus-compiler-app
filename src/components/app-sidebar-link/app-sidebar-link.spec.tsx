import { shallow } from 'enzyme'
import React from 'react'
import { NavLink } from 'react-router-dom'
import AppSidebarLink from './app-sidebar-link'

describe('<AppSidebarLink>', () => {
  it('renders correctly', () => {
    const component = (
      <AppSidebarLink to="/toto" />
    )

    expect(component).toBeTruthy()
  })

  it('matches snapshot when no exact', () => {
    const component = shallow((
      <AppSidebarLink to="/toto" />
    ))

    expect(component).toMatchSnapshot()

    const navLink = component.find(NavLink)

    expect(navLink).toHaveLength(1)
    expect(navLink.prop('to')).toBe('/toto')
    expect(navLink.prop('exact')).toBe(undefined)
  })

  it('matches snapshot when exact', () => {
    const component = shallow((
      <AppSidebarLink to="/tata" exact />
    ))

    expect(component).toMatchSnapshot()

    const navLink = component.find(NavLink)

    expect(navLink).toHaveLength(1)
    expect(navLink.prop('to')).toBe('/tata')
    expect(navLink.prop('exact')).toBe(true)
  })
})
