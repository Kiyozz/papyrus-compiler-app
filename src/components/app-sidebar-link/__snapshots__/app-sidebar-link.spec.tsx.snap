// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSidebarLink> matches snapshot when exact 1`] = `
<NavLink
  activeClassName="app-sidebar-link-active"
  className="app-sidebar-link"
  exact={true}
  onClick={[Function]}
  to="/tata"
/>
`;

exports[`<AppSidebarLink> matches snapshot when no exact 1`] = `
<NavLink
  activeClassName="app-sidebar-link-active"
  className="app-sidebar-link"
  onClick={[Function]}
  to="/toto"
/>
`;
